packages/back-end/src/services/stripe.ts

```diff
@@ -0,0 +1,66 @@
+import { STRIPE_SECRET } from "../util/secrets";
+import { Stripe } from "stripe";
+import { updateOrganizationByStripeId } from "../models/OrganizationModel";
+const stripe = new Stripe(STRIPE_SECRET || "", { apiVersion: "2020-08-27" });
+
+/**
+ * @name updateSubscriptionInDb
+ * @description This function updates the subscription in the database. (organization.subscription)
+ */
+export async function updateSubscriptionInDb(
+  subscription: string | Stripe.Subscription
+) {
+  // Make sure we have the full subscription object
+  if (typeof subscription === "string") {
+    subscription = await stripe.subscriptions.retrieve(subscription, {
+      expand: ["plan"],
+    });
+  }
+
+  const stripeCustomerId =
+    typeof subscription.customer === "string"
+      ? subscription.customer
+      : subscription.customer.id;
+
+  await updateOrganizationByStripeId(stripeCustomerId, {
+    subscription: {
+      id: subscription.id,
+      qty: subscription.items.data[0].quantity || 1,
```

Comment:
All of these deeply nested properties could cause exceptions if they don't exist.  Might be better to use optional chaining (`a?.b?.[0]?.c`)

Reply:
Updated. Let me know if this is sufficient.